packages/babel-parser/src/parser/expression.js

```diff
@@ -1934,14 +1934,33 @@ export default class ExpressionParser extends LValParser {
       const oldLabels = this.state.labels;
       this.state.labels = [];
       if (useStrict) this.state.strict = true;
```

Comment:
This (and the whole old `useStrict` logic) can be removed. You only need to keep the labels logic.